warehouse/query-core/src/main/java/datawave/query/common/grouping/GroupingUtil.java

```diff
@@ -172,7 +172,9 @@ private Multimap<String,String> getFieldToFieldWithGroupingContextMap(Document d
                 log.trace("{} contains {}", groupFieldsSet, shorterName);
```

Comment:
Looking at this, lines 158-159 are equivalent to what was done between 153-154 because shortName and shorterName are initialized to be the same value.  I am wondering if we are missing a statement at line 154 reassigning shorterName to shortName.  I wonder if this oversight was part of the problem.

Reply:
Are shortName and shorterName ever different?  Doesn't look like it to me.

Reply:
Yes, they can be.  Cases where fieldname is of the form FIELD_NAME.GROUP.{other info}. GROUPING_CONTEXT.

The problem this is trying to address is for processor-generated fields where we didn't increase the grouping context counter (this should be fixed too!)  It causes a collision in the map so we only ended up returning one instance of the field. 

This class is overdue for an overhaul in any case.  Things could definitely be simplified, but I propose we address this bug first and refactor in a separate ticket.

Reply:
Unit test was added to cover case described by @hlgp.